### ISSUE TITLE
Suspect controller type in MixedAir::UpdateOAController when contaminant simulation is active

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#5535]
## Issue overview

Noticed these lines of code in MixedAir::UpdateOAController while working on #5483:

```
        if ( Contaminant.CO2Simulation && OAController( OAControllerNum ).ControllerType_Num == ControllerSimple ) {
            Node( RelAirNodeNum ).CO2 = Node( InletAirNodeNum ).CO2;
            if ( Node( RetAirNodeNum ).MassFlowRate > 0.0 ) {
                Node( RetAirNodeNum ).CO2 = ( ( Node( InletAirNodeNum ).MassFlowRate - Node( RelAirNodeNum ).MassFlowRate ) * Node( InletAirNodeNum ).CO2 + Node( OutAirNodeNum ).MassFlowRate * OutdoorCO2 ) / Node( RetAirNodeNum ).MassFlowRate;
            } else {
                Node( RetAirNodeNum ).CO2 = Node( InletAirNodeNum ).CO2;
            }
        }

        if ( Contaminant.GenericContamSimulation && OAController( OAControllerNum ).ControllerType_Num == ControllerSimple ) {
            Node( RelAirNodeNum ).GenContam = Node( InletAirNodeNum ).GenContam;
            if ( Node( RetAirNodeNum ).MassFlowRate > 0.0 ) {
                Node( RetAirNodeNum ).GenContam = ( ( Node( InletAirNodeNum ).MassFlowRate - Node( RelAirNodeNum ).MassFlowRate ) * Node( InletAirNodeNum ).GenContam + Node( OutAirNodeNum ).MassFlowRate * OutdoorGC ) / Node( RetAirNodeNum ).MassFlowRate;
            } else {
                Node( RetAirNodeNum ).GenContam = Node( InletAirNodeNum ).GenContam;
            }
        }
```

These block only execute if `OAController( OAControllerNum ).ControllerType_Num == ControllerSimple`, but there are no OA controllers of that type.  The only valid types are ControllerOutsideAir and ControllerStandAloneERV.

@lgu1234 Looks like these should be checking for `OAController( OAControllerNum ).ControllerType_Num == ControllerOutsideAir`

A check of the [integration test code coverage](http://energyplus.s3-website-us-east-1.amazonaws.com/coverage/2016-03-07-EnergyPlus-ffaabe11cd-x86_64-gcc-4.8ICD/EnergyPlus/MixedAir.cc.gcov.html) shows that these lines are never executed in the test suite.

![image](https://cloud.githubusercontent.com/assets/4991075/13611110/79533290-e526-11e5-93cf-e00b7baece6b.png)
### Details
- Develop as of today https://github.com/NREL/EnergyPlus/commit/ffaabe11cd573fa5493636d9926a177878396c3d
### Checklist
- [ ] Defect file added (list location of defect file here) - _any test file with CO2 or Generic contaminant controls active_
- [ ] Ticket added to Pivotal for defect (development team task)
- [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by lgu1234** [comment]
@mjwitte You are right. Let me fix it and run a local test.

**Comment by mjwitte** [comment]
@lgu1234 Thanks.  I would also suggest deleting the ControllerSimple declaration in this module.

**Comment by lgu1234** [comment]
@mjwitte In addition, I also found two sections in SizeOAController. For example as a section:

```
            { auto const SELECT_CASE_var( OAController( OAControllerNum ).ControllerType_Num );

            if ( SELECT_CASE_var == ControllerOutsideAir ) {

                CheckZoneSizing( CurrentModuleObject, OAController( OAControllerNum ).Name );
                OAController( OAControllerNum ).MaxOA = max( FinalZoneSizing( CurZoneEqNum ).DesCoolVolFlow, FinalZoneSizing( CurZoneEqNum ).DesHeatVolFlow );

            } else if ( SELECT_CASE_var == ControllerSimple ) {

            } else if ( SELECT_CASE_var == ControllerStandAloneERV ) {

            } else {

            }}
```

Therefore, I will delete additional two lines with ControllerSimple.

**Comment by lgu1234** [comment]
@mjwitte After careful debug, I found the suspected sections are not needed. Actually, they are wrong actions. Here is why.

The suspected sections determine CO2 levels at relief node and return node. The return node CO2 level is determined by a AirLoopHVAC:ZoneMixer. No need to make any modifications. The relief node CO2 level is determined in UpdateOAMixer:
            Node( RelNode ).CO2 = Node( RetNode ).CO2;
This is the same process as temperature.

The above description is also true to contaminant.

Then I run an example file of HeatPumpIAQP_DCV.idf by adding 3 more output variables:

  Output:Variable,*,System Node CO2 Concentration,timestep;

  Output:Variable,*,System Node Mass Flow Rate,timestep;

  Output:Variable,*,System Node Temperature,timestep;

I checked values of 4 nodes for an OA mixer and found results are expected.

After deleting the suspected sections, no differences were found in 4 example input files using the ZoneAirContaminantBalance object.

My suggestion is to delete the suspected sections and cleanup the ControllerSimple declaration.

**Comment by mjwitte** [comment]
@lgu1234 Sounds good.

**Comment by lgu1234** [comment]
@mjwitte Created a pull request and added a test file in EnergyPlusDevSupport

**Comment by mjwitte** [comment]
Addressed via #5538
